build-packages/webpack-i18n-runtime/build-config/webpack-i18n-tracker/localization-manager/index.js

```diff
@@ -0,0 +1,208 @@
+        createMissingTranslationFiles(this.enabledLocales, this.defaultLocale);
+    }
+
+    unlinkTranslatablesFromModule(moduleId) {
+        if (!this.moduleMap[moduleId]) {
+            return;
+        }
+
+        this.moduleMap[moduleId].clear();
+    }
+
+    linkTranslatableToModule(moduleId, translatable) {
+        if (!this.moduleMap[moduleId]) {
+            this.moduleMap[moduleId] = new Set();
+        }
+
+        this.moduleMap[moduleId].add(translatable);
+    }
+
+    updateUsedTranslations() {
+        // Generate an array of used translations
+        const usedTranslationsArray = Object.values(this.moduleMap)
+            .reduce(
+                (acc, cur) => acc.concat(...cur),
+                []
+            );
+
+        // Ensure uniqueness for them
+        this.usedTranslations = new Set(usedTranslationsArray);
+    }
+
+    /**
+     * Handle missing translations not yet present in corresponding files
+     */
+    handleMissingTranslations() {
+        // Initialize missing translations
+        const missingTranslations = {};
+
+        // Check each translatable
+        for (const usedTranslation of this.usedTranslations) {
+            // Handle locales separately
+            for (const localeCode in this.translationMap) {
+                // Handle missing translation
+                if (!this.translationMap[localeCode].merged[usedTranslation]) {
+                    // Handle first missing translation for locale
+                    if (!missingTranslations[localeCode]) {
+                        missingTranslations[localeCode] = new Set();
+                    }
+
+                    missingTranslations[localeCode].add(usedTranslation);
+                }
+            }
+        }
+
+        // Ignore the default locale
+        delete missingTranslations[this.defaultLocale];
+
+        // Count the missing translations for each locale
+        const missingTranslationsCountMap = Object.fromEntries(Object.entries(missingTranslations).map(
+            ([localeCode, missingTranslations]) => [localeCode, missingTranslations.size]
+        ));
+
+        // Handle OK case
+        if (!Object.keys(missingTranslationsCountMap).length) {
+            return;
+        }
+
+        // 1. log which ones are missing
+        afterEmitLogger.logMessage(missingTranslationsMessage(missingTranslationsCountMap));
+
+        // 2. Update files contents
+        appendMissingTranslationsToFiles(missingTranslations);
+    }
+
+    /**
+     * Handle unused translations
+     */
+    handleUnusedTranslations() {
+        const unusedTranslations = {};
+
+        // Loop over translations from files
+        for (const localeCode in this.translationMap) {
+            // Check each translation from there
+            for (const translatable in this.translationMap[localeCode].child) {
+                // Handle translation not used in the application
+                if (!this.usedTranslations.has(translatable)) {
```

Comment:
Can we reverse these ifs? Nesting seem too huge, the early return might help :)

Reply:
We can, if `continue` statements will not scare people :)

Reply:
It is fine